src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -214,907 +214,908 @@ void GetPurchasedAir(EnergyPlusData &state)
-            PurchAir(PurchAirNum).LatHeatRate = 0.0;
-            PurchAir(PurchAirNum).TotHeatRate = 0.0;
-            PurchAir(PurchAirNum).SenCoolRate = 0.0;
-            PurchAir(PurchAirNum).LatCoolRate = 0.0;
-            PurchAir(PurchAirNum).TotCoolRate = 0.0;
-            PurchAir(PurchAirNum).ZoneSenHeatRate = 0.0;
-            PurchAir(PurchAirNum).ZoneLatHeatRate = 0.0;
-            PurchAir(PurchAirNum).ZoneTotHeatRate = 0.0;
-            PurchAir(PurchAirNum).ZoneSenCoolRate = 0.0;
-            PurchAir(PurchAirNum).ZoneLatCoolRate = 0.0;
-            PurchAir(PurchAirNum).ZoneTotCoolRate = 0.0;
-            PurchAir(PurchAirNum).OASenHeatRate = 0.0;
-            PurchAir(PurchAirNum).OALatHeatRate = 0.0;
-            PurchAir(PurchAirNum).OATotHeatRate = 0.0;
-            PurchAir(PurchAirNum).OASenCoolRate = 0.0;
-            PurchAir(PurchAirNum).OALatCoolRate = 0.0;
-            PurchAir(PurchAirNum).OATotCoolRate = 0.0;
-            PurchAir(PurchAirNum).HtRecSenHeatRate = 0.0;
-            PurchAir(PurchAirNum).HtRecLatHeatRate = 0.0;
-            PurchAir(PurchAirNum).HtRecTotHeatRate = 0.0;
-            PurchAir(PurchAirNum).HtRecSenCoolRate = 0.0;
-            PurchAir(PurchAirNum).HtRecLatCoolRate = 0.0;
-            PurchAir(PurchAirNum).HtRecTotCoolRate = 0.0;
-
-            PurchAir(PurchAirNum).OutdoorAirMassFlowRate = 0.0;
-            PurchAir(PurchAirNum).OutdoorAirVolFlowRateStdRho = 0.0;
-            PurchAir(PurchAirNum).SupplyAirMassFlowRate = 0.0;
-            PurchAir(PurchAirNum).SupplyAirVolFlowRateStdRho = 0.0;
+            // get optional heating fuel efficiency schedule name
+            cAlphaFieldName = "Heating Fuel Efficiency Schedule Name";
+            std::string const heatFuelEfficiencySchedName = s_ip->getAlphaFieldValue(fields, schemaProps, "heating_fuel_efficiency_schedule_name");
+            if (heatFuelEfficiencySchedName.empty()) {
+                PurchAir.heatFuelEffSched = Sched::GetScheduleAlwaysOn(state);
+            } else if ((PurchAir.heatFuelEffSched = Sched::GetSchedule(state, heatFuelEfficiencySchedName)) == nullptr) {
+                ShowSevereItemNotFound(state, eoh, cAlphaFieldName, heatFuelEfficiencySchedName);
+                ErrorsFound = true;
+            }
+            // get optional cooling fuel efficiency schedule name
+            cAlphaFieldName = "Cooling Fuel Efficiency Schedule Name";
+            std::string const coolFuelEfficiencySchedName = s_ip->getAlphaFieldValue(fields, schemaProps, "cooling_fuel_efficiency_schedule_name");
+            if (coolFuelEfficiencySchedName.empty()) {
+                PurchAir.coolFuelEffSched = Sched::GetScheduleAlwaysOn(state);
+            } else if ((PurchAir.coolFuelEffSched = Sched::GetSchedule(state, coolFuelEfficiencySchedName)) == nullptr) {
+                ShowSevereItemNotFound(state, eoh, cAlphaFieldName, coolFuelEfficiencySchedName);
+                ErrorsFound = true;
+            }
```

Comment:
Reading in the `Cooling Fuel Efficiency Schedule Name` optional new input field. If this field is blank, AlwaysOn constant schedule is assigned, which has a constant schedule value of 1.